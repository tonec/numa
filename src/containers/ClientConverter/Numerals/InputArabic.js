import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Input } from 'antd'
import { isNumber } from 'helpers'

export class InputArabic extends Component {

  static propTypes = {
    value: PropTypes.string,
    updateValue: PropTypes.func.isRequired
  }

  static defaultProps = {
    value: null
  }

  handleOnInputChange = e => {
    const { value } = e.target

    if (value > 3999) return

    if (isNumber(value) || value.length === 0) {
      this.props.updateValue(value)
    }
  }

  render () {
    return (
      <Input
        size="large"
        placeholder="Arabic"
        value={this.props.value}
        onChange={this.handleOnInputChange}
      />
    )
  }
}

export default InputArabic
