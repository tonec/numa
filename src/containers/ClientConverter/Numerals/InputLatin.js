import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Input } from 'antd'
import { isLatinChar } from 'helpers'

export class InputLatin extends Component {

  static propTypes = {
    value: PropTypes.string,
    updateValue: PropTypes.func.isRequired
  }

  static defaultProps = {
    value: null
  }

  handleInputChange = e => {
    const { value } = e.target

    if (isLatinChar(value) || value.length === 0) {
      this.props.updateValue(value.toUpperCase())
    }
  }

  render () {
    return (
      <Input
        size="large"
        placeholder="Latin"
        value={this.props.value}
        onChange={this.handleInputChange}
      />
    )
  }
}

export default InputLatin
