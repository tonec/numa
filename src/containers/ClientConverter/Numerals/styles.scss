
.iconContainer {
  text-align: center;

  i {
    transform: rotate(-45deg);
  }
}