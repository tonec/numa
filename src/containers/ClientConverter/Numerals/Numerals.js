import React, { Component } from 'react'
import { Icon } from 'antd'
import InputArabic from './InputArabic'
import InputLatin from './InputLatin'
import { toLatin, fromLatin } from 'helpers/convertNumerals'

import styles from './styles.scss'

export class Numerals extends Component {

  constructor (props) {
    super(props)

    this.state = {
      arabicValue: null,
      latinValue: null
    }
  }

  handleUpdateArabicValue = val => {
    this.setState({ arabicValue: val })
    this.updateAndConvertLatinValue(val)
  }

  handleUpdateLatinValue = val => {
    this.setState({ latinValue: val.toUpperCase() })
    this.updateAndConvertArabicValue(val)
  }

  updateAndConvertArabicValue = val => {
    this.setState({ arabicValue: fromLatin(val) ? fromLatin(val).toString() : null })
  }

  updateAndConvertLatinValue = val => {
    this.setState({ latinValue: toLatin(val) })
  }

  render () {
    const { arabicValue, latinValue } = this.state

    return (
      <div>
        <InputArabic value={arabicValue} updateValue={this.handleUpdateArabicValue} />
        <div className={styles.iconContainer}><Icon type="arrows-alt" /></div>
        <InputLatin value={latinValue} updateValue={this.handleUpdateLatinValue} />
      </div>
    )
  }
}

export default Numerals
