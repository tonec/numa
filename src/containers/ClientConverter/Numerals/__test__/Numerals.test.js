import React from 'react'
import { shallow } from 'enzyme'
import { Numerals } from '../Numerals'

describe('Numerals (snapshot)', () => {
  it('renders according to snapshot', () => {
    const component = shallow(<Numerals />)
    expect(component).toMatchSnapshot()
  })
})
