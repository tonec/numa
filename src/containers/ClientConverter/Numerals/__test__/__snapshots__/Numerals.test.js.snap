// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numerals (snapshot) renders according to snapshot 1`] = `
<OneColumnThin
  className=""
  offset={9}
  span={6}
  title="Numerals"
>
  <Input
    disabled={false}
    placeholder="Arabic"
    prefixCls="ant-input"
    size="large"
    type="text"
    value={null}
  />
  <Input
    disabled={false}
    placeholder="Latin"
    prefixCls="ant-input"
    size="large"
    type="text"
    value={null}
  />
</OneColumnThin>
`;
